src/main/java/com/google/devtools/build/lib/skyframe/toolchains/ToolchainResolutionFunction.java

```diff
@@ -80,14 +80,7 @@ public UnloadedToolchainContext compute(SkyKey skyKey, Environment env)
           Preconditions.checkNotNull(configuration.getFragment(PlatformConfiguration.class));
 
       // Check if debug output should be generated.
-      boolean debug =
-          key.debugTarget()
```

Comment:
The `--toolchain_resolution_debug=regex` flag has two usages:
1. If the regex matches a toolchain type, all attempts to resolve that toolchain type are logged.
   1. This is the most common usage, and we have several help messages telling users to do this.
2. If the regex matches a target that requests a toolchain type, toolchain resolution for that target is logged.
   1. This is what causes the `debugTarget` field on the key to be set.
 
This change is removing the first part: why was it removed?

Reply:
There are two kinds of debug messages
A. A debug trace of the toolchain resolution for a given toolchain type, and a given target platform. These are multi-line traces.
B. single line target platform selection, based on the first target platform that has all the required toolchain types.

I did it to remove verbosity in case 1. where presumably all the B. messages are less useful and spammy. It seems more useful to get B messages for a given target than for a given toolchain type. B messages are also somewhat redundant, because A. messages already tell which toolchain was selected for which target platform.

So getting all the B messages seems somewhat verbose. Also, the same behavior can be achieved by grepping the toolchain type in the` --toolchain_resolution_debug=.*` output

All in all, because I felt it was better.

I did not realise that such a debug feature was used extensively in tests.
You can see that I had to adapt the tests to use case 2. to recover B. messages that my PR removed in case 1.

Reply:
Using the flag with a specific toolchain type is unfortunately more verbose, but is definitely something that users expect to be able to do, and we should have a longer planning period before removing that. Please add it back in.

Reply:
Reverted !